android/src/main/java/com/survicate/react/SurvicateModuleImpl.java

```diff
@@ -67,4 +71,27 @@ public void reset() {
     public void setWorkspaceKey(String workspaceKey) {
         Survicate.setWorkspaceKey(workspaceKey);
     }
+
+    @ReactMethod
+    public void addListener(String eventName) {
+        if (listenerCount == 0) {
+            Survicate.setEventListener(eventListener);
+        }
+
+        listenerCount++;
+    }
+
+    @ReactMethod
+    public void removeListeners(int count) {
+        listenerCount -= count;
+
+        if (listenerCount == 0) {
+            Survicate.setEventListener(null);
+        }
+    }
+
+    @ReactMethod
+    public void removeListeners(double count) {
+        removeListeners((int) count);
+    }
```

Comment:
What's the use case of removing by count? It's preparation for native SDK update?

Reply:
The way the react native listeners work is that each method is listened independently from another. I am adding one listener per method (on react native it looks like this addListener("onSurveyDisplayed", [...])). So each addListener call will require the same amount of listeners to be removed